h2o-core/src/main/java/water/H2O.java

```diff
@@ -1915,6 +1935,13 @@ public static void main( String[] args ) {
     // join an existing Cloud.
     new HeartBeatThread().start();
 
+    // register disconnect hooks for the disconnect watchdog
+    registerNodeDisconnectedHook(new NodeDisconnectHook.ClientDisconnectedHook());
+
+    // Thread which periodically checks the state node of nodes and executes all disconnect hooks in case
+    // the node was disconnected
+    new DisconnectedNodesWatchdog().start();
```

Comment:
I'm also thinking of renaming the `Disconnected` to `Failed` or anything similar. Because otherwise the `disconnect` event would have 2 meanings - client event disconnect and failed node